allensdk/brain_observatory/behavior/behavior_project_cache/behavior_ecephys_project_cache.py

```diff
@@ -0,0 +1,283 @@
+        if not isinstance(self.fetch_api, BehaviorEcephysProjectCloudApi):
+            self._cache_not_implemented('latest_downloaded_manifest_file')
+        return self.fetch_api.cache.latest_downloaded_manifest_file
+
+    def latest_manifest_file(self) -> str:
+        """
+        Return the name of the most up to date data manifest
+        corresponding to this dataset, checking in the cloud
+        if this is a cloud-backed cache.
+        """
+        if not isinstance(self.fetch_api, BehaviorEcephysProjectCloudApi):
+            self._cache_not_implemented('latest_manifest_file')
+        return self.fetch_api.cache.latest_manifest_file
+
+    def load_manifest(self, manifest_name: str):
+        """
+        Load a specific versioned manifest for this dataset.
+
+        Parameters
+        ----------
+        manifest_name: str
+            The name of the manifest to load. Must be an element in
+            self.manifest_file_names
+        """
+        if not isinstance(self.fetch_api, BehaviorEcephysProjectCloudApi):
+            self._cache_not_implemented('load_manifest')
+        self.fetch_api.load_manifest(manifest_name)
+
+    def list_all_downloaded_manifests(self) -> list:
+        """
+        Return a sorted list of the names of the manifest files
+        that have been downloaded to this cache.
+        """
+        if not isinstance(self.fetch_api, BehaviorEcephysProjectCloudApi):
+            self._cache_not_implemented('list_all_downloaded_manifests')
+        return self.fetch_api.cache.list_all_downloaded_manifests()
+
+    def list_manifest_file_names(self) -> list:
+        """
+        Return a sorted list of the names of the manifest files
+        associated with this dataset.
+        """
+        if not isinstance(self.fetch_api, BehaviorEcephysProjectCloudApi):
+            self._cache_not_implemented('list_manifest_file_names')
+        return self.fetch_api.cache.manifest_file_names
+
+    def current_manifest(self) -> Union[None, str]:
+        """
+        Return the name of the dataset manifest currently being
+        used by this cache.
+        """
+        if not isinstance(self.fetch_api, BehaviorEcephysProjectCloudApi):
+            self._cache_not_implemented('current_manifest')
+        return self.fetch_api.cache.current_manifest
+
+    def get_ecephys_session_table(self):
+        return self.fetch_api.get_ecephys_session_table(),
+
+    def get_behavior_session_table(self):
+        return self.fetch_api.get_behavior_session_table(),
+
+    def get_probe_table(self):
+        self.fetch_api.get_probe_table(),
+
+    def get_channel_table(self):
+        return self.fetch_api.get_channel_table(),
+
+    def get_unit_table(self):
+        return self.fetch_api.get_unit_table(),
+
+    def get_ecephys_session(self, ecephys_session_id: int):
+
+        fetch_session = partial(self.fetch_api.get_ecephys_session,
+                                ecephys_session_id)
+        return call_caching(
```

Comment:
I'm not sure we need to use `call_caching`.

My understanding is that it is a piece of legacy code in the VBO cache that we did not want to wantonly throw out. Looking at the definition of `call_caching`, it is poorly documented and hard to follow. I would rather not introduce new dependencies on it.

I'm also not sure what it is getting us. These session objects can be quite large, and I don't think we should be silently building an on-disk cache of them without the users knowing that we are doing so. Unless there is a clear advantage to calling `call_cache` that we are trying to get, I think `get_ecephys_session` should just return the result of `fetch_api.get_ecephys_session` without passing through `call_caching`.